app/src/chf/java/org/openimis/imispolicies/Escape.java

```diff
@@ -0,0 +1,60 @@
+//Copyright (c) 2016-%CurrentYear% Swiss Agency for Development and Cooperation (SDC)
+//
+//The program users must agree to the following terms:
+//
+//Copyright notices
+//This program is free software: you can redistribute it and/or modify it under the terms of the GNU AGPL v3 License as published by the 
+//Free Software Foundation, version 3 of the License.
+//This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of 
+//MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU AGPL v3 License for more details www.gnu.org.
+//
+//Disclaimer of Warranty
+//There is no warranty for the program, to the extent permitted by applicable law; except when otherwise stated in writing the copyright 
+//holders and/or other parties provide the program "as is" without warranty of any kind, either expressed or implied, including, but not 
+//limited to, the implied warranties of merchantability and fitness for a particular purpose. The entire risk as to the quality and 
+//performance of the program is with you. Should the program prove defective, you assume the cost of all necessary servicing, repair or correction.
+//
+//Limitation of Liability 
+//In no event unless required by applicable law or agreed to in writing will any copyright holder, or any other party who modifies and/or 
+//conveys the program as permitted above, be liable to you for damages, including any general, special, incidental or consequential damages 
+//arising out of the use or inability to use the program (including but not limited to loss of data or data being rendered inaccurate or losses 
+//sustained by you or third parties or a failure of the program to operate with any other programs), even if such holder or other party has been 
+//advised of the possibility of such damages.
+//
+//In case of dispute arising out or in relation to the use of the program, it is subject to the public law of Switzerland. The place of jurisdiction is Berne.
+
+package org.openimis.imispolicies;
+
+/**
+ * Created by Hiren on 12/05/2017.
+ */
+
+public class Escape {
+    public int CheckInsuranceNumber(String InsuranceNumber){
+
+        if (InsuranceNumber.length() == 0){
+            return R.string.MissingInsuranceNumber;
+        }
+
+        if (!isValidInsuranceNumber(InsuranceNumber)){
+            return R.string.InvalidInsuranceNumber;
+
+        }
+
+        return 0;
+    }
+
+    private boolean isValidInsuranceNumber(String InsuranceNumber){
+        if (InsuranceNumber.length() != 9) return false;
+        String chfid;
+        int Part1, Part2;
+        Part1 = Integer.parseInt(InsuranceNumber.toString())/10;
+        Part2 = Part1 % 7;
+
+        chfid = InsuranceNumber.substring(0, 8) + Integer.toString(Part2);
```

Comment:
Why using 8 first digits in one place (l54) and the removing the last digit in another place (l51)?
Also I Think it will make more sense to return InsuranceNumber.substring(9).equals(Part2);

Reply:
It's validation for CHF-IMIS. Ninth digit is modulo 7 of the number represented by the first eight digits.
I've updated method to improve readability